src/plugin/ipc/file/fileconnection.cpp

```diff
@@ -351,9 +351,17 @@ FileConnection::refill(bool isRestart)
           // source at restart, the requests are also saved after checkpoint.
           // The log file size at restart is larger than its size saved in the
           // checkpoint image. Just give a warning here and continue.
-          JWARNING(false) (_path) (_st_size) (statbuf.st_size)
-          .Text("Setting saved size to the current file size");
-          _st_size = statbuf.st_size;
+          if (dmtcp_should_truncate_file_on_restart(_path.c_str())) {
```

Comment:
It should be another way around. By default, DMTCP should truncate and it should avoid truncating only when the plugin instructs it to avoid.